src/internet_identity_interface/src/lib.rs

```diff
@@ -173,3 +173,92 @@ pub struct HttpResponse {
 pub struct InternetIdentityInit {
     pub assigned_user_number_range: (UserNumber, UserNumber),
 }
+
+// Archive specific types
+
+#[derive(Eq, PartialEq, Clone, Debug, CandidType, Deserialize)]
+pub enum OperationType {
+    #[serde(rename = "register_anchor")]
+    RegisterAnchor {
+        initial_device: DeviceDataWithoutAlias,
+    },
+    #[serde(rename = "add_device")]
+    AddDevice { new_device: DeviceDataWithoutAlias },
+    #[serde(rename = "update_device")]
+    UpdateDevice {
+        updated_device: PublicKey,
+        changed_data: DeviceDataUpdate,
+    },
+    #[serde(rename = "remove_device")]
+    RemoveDevice { removed_device: PublicKey },
+}
+
+#[derive(Eq, PartialEq, Clone, Debug, CandidType, Deserialize)]
+pub struct LogEntry {
+    // store user_number in LogEntry, such that anchor operations can be attributed to a user without consulting the index.
+    pub user_number: UserNumber,
+    pub operation: OperationType,
+    pub timestamp: Timestamp,
+    pub caller: Principal,
+    // global sequence number to detect lost messages (if any)
+    pub sequence_number: u64,
+}
+
+#[derive(Eq, PartialEq, Clone, Debug, CandidType, Deserialize)]
+pub struct DeviceDataWithoutAlias {
+    pub pubkey: DeviceKey,
+    pub credential_id: Option<CredentialId>,
+    pub purpose: Purpose,
+    pub key_type: KeyType,
+    pub protection: DeviceProtection,
+}
+
+// If present, the attribute has been changed to the value given.
+// Does not include the pubkey because it cannot be changed.
+#[derive(Eq, PartialEq, Clone, Debug, CandidType, Deserialize)]
+pub struct DeviceDataUpdate {
+    pub alias: Option<Hidden>,
+    pub credential_id: Option<CredentialId>,
+    pub purpose: Option<Purpose>,
+    pub key_type: Option<KeyType>,
+    pub protection: Option<DeviceProtection>,
+}
+
+// Placeholder for information that has been hidden for privacy reasons.
+#[derive(Eq, PartialEq, Clone, Debug, CandidType, Deserialize)]
+pub enum Hidden {
+    HiddenForPrivacyReasons,
+}
+
+#[derive(Clone, Debug, CandidType, Deserialize)]
+pub struct Logs {
+    // make this a vec of options to keep LogEntry extensible
```

Comment:
why does this keep `LogEntry` extensible? if it's a matter of wrapping it in an enum, can we make a single-constructor enum with an appropriate name?